pkg/kubectl/cmd/log.go

```diff
@@ -137,28 +141,45 @@ func RunLog(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 		}
 	}
 
-	follow := false
-	if cmdutil.GetFlagBool(cmd, "follow") {
-		follow = true
+	logOptions := &api.PodLogOptions{
+		Follow:     cmdutil.GetFlagBool(cmd, "follow"),
+		Previous:   cmdutil.GetFlagBool(cmd, "previous"),
+		Timestamps: cmdutil.GetFlagBool(cmd, "timestamps"),
+		Since:      cmdutil.GetFlagString(cmd, "since"),
 	}
-
-	previous := false
-	if cmdutil.GetFlagBool(cmd, "previous") {
-		previous = true
+	if limitBytes := cmdutil.GetFlagInt(cmd, "limit-bytes"); limitBytes != 0 {
```

Comment:
technically should be converting to int64 from the flag... pflag does have a GetInt64 method... a new method can be added to cmdutil to call it.